memiavl/db.go

```diff
@@ -341,6 +354,19 @@ func (db *DB) checkBackgroundSnapshotRewrite() error {
 			return fmt.Errorf("background snapshot rewriting failed: %w", result.err)
 		}
 
+		// wait for potential pending wal writings to finish, to make sure we catch up to latest state.
+		// in real world, block execution should be slower than wal writing, so this should not block for long.
+		for {
+			committedVersion, err := db.CommittedVersion()
+			if err != nil {
+				return fmt.Errorf("get wal version failed: %w", err)
+			}
+			if db.lastCommitInfo.Version == committedVersion {
+				break
+			}
+			time.Sleep(time.Nanosecond)
```

Comment:
LGTM, but 1ns is quite frequent.

Reply:
The purpose is to reduce blocking time in Commit as much as possible, the case for the loop to happen should be very rare in real world, it has never happened on production so far.
But I think we might need some timeout control here, in case wal writing thread crashed somehow, we'll be blocked indefinitely.

Reply:
Let's fix the bug first